### DISCUSSION TITLE
How to enable access to "git lfs" for a vcpkg port?

### DISCUSSION THREAD

**Discussion by trots** [microsoft/vcpkg#29064] [Q&A] [answered]
In my overlay port I'm trying to get sources from GitHub repo with `LFS` attribute enabled:
```cmake
vcpkg_from_git(
  OUT_SOURCE_PATH SOURCE_PATH
  URL git@github.com:some/repo.git
  REF c8f92ce7110fffcd0d8c34c32c1f5ab744c7a731
  LFS)
```
And I get the error message: `Git LFS is required for myport`. But Git LFS is installed on the system and is available from the command line.
I spent a lot of time figuring out the cause of this error and found that `vcpkg` re-initializes all environment variables before starting a port. As a result, Git LFS becomes inaccessible from `vcpkg`.

Could you please help me how to use the Git LFS from a `vcpkg` port properly?

**Comment by Osyotr** [comment, marked as answer]
If Git LFS does not work because some env variable is not passed into build, you may want to add this variable to `VCPKG_ENV_PASSTHROUGH_UNTRACKED`. https://learn.microsoft.com/en-us/vcpkg/users/triplets#vcpkg_env_passthrough_untracked

**Reply by trots** [reply]
Thank you, it works. I added this line into my triplet file:
```
set(VCPKG_ENV_PASSTHROUGH GIT_LFS_PATH)
```
And then in the port script I added the `GIT_LFS_PATH` value to the `PATH` variable:
```
set(ENV{PATH} "$ENV{GIT_LFS_PATH};$ENV{PATH}")
```

**Reply by Osyotr** [reply]
You may drop this workaround after https://github.com/microsoft/vcpkg/pull/28693 is merged

**Comment by retifrav** [comment]
For those who will land here from Google, I've got the same error (*"Git LFS is required for ..."*) on one of my Windows buildbots, and turned out that it had Git v2.33 installed there, which is somewhat old, and so I updated it to v2.41, and then Git LFS operation succeeded. If anything, the Git LFS package is v3.3.0.

So in my case the problem was not environment variables but "old" version of Git.

**Reply by retifrav** [reply]
...some time later I got the same problem on a different machine, and there both Git and Git LFS were already the latest versions.

And then [this workaround](https://github.com/microsoft/vcpkg/discussions/29064#discussioncomment-4757449) helped, although I've set it in my triplet file instead of a single port, and not `GIT_LFS_PATH` but just path to `git-lfs.exe` folder.

That made me suspect that there is something wrong with the environment variables, and indeed, turned out user was working in Git BASH, and he did some environment manipulations in his `~/.bash_profile`, messing up environment variables set in Windows settings, which is why appending `git-lfs.exe` folder to `PATH` in triplet helped to discover it. So the solution in this case was to clean up user's `~/.bash_profile`, so vcpkg/CMake would get `git-lfs` executable location from `PATH` (*and naturally then there was no need to modify the triplet anymore*).

**Reply by Neumann-A** [reply]
So the TL;DR is just dont mess with your settings or be at least aware of the implications?

**Reply by retifrav** [reply]
Depends on the case. In first one it was about old Git version, in second one it was about messed up environment variables. I described both for the benefit of those who'll stumble upon this thread when googling for that error message.
